### PR TITLE
Update backbone.js

### PR DESCRIPTION
**Pull request by zgl2007dj**
1 ) improve the patch method
 2 ) _.size is not nessesary, bool will be better

### CODE REVIEW

File: backbone.js [hunk 1]
```diff
@@ -644,7 +646,7 @@
       if (attrs && wait) this.attributes = _.extend({}, attributes, attrs);
 
       var method = this.isNew() ? 'create' : (options.patch ? 'patch' : 'update');
-      if (method === 'patch' && !options.attrs) options.attrs = attrs;
+      if (method === 'patch' && !options.attrs) options.attrs = this.changedAttributes(attrs);
```
Comment (akre54) [inline backbone.js]: That's not how this method is supposed to work but the rest of the pull is great. Back out this change, and I'll merge. Thanks!

Comment (zgl2007dj) [inline backbone.js]: Thanks, I fixed it.
 But,  i want the method work like that, and overrided model's set function in my app. Maybe not the best way : )

Comment (akre54) [inline backbone.js]: You can override `Model#save` or `Model#sync` to add this in your own app if you need it.

**Comment by akre54** [comment]
Can you squash down into a single commit and I'll merge. Thanks

File: backbone.js [hunk 2]
```diff
@@ -562,12 +562,14 @@
       if (!diff) return this.hasChanged() ? _.clone(this.changed) : false;
       var old = this._changing ? this._previousAttributes : this.attributes;
       var changed = {};
+      var hasChanged;
       for (var attr in diff) {
         var val = diff[attr];
         if (_.isEqual(old[attr], val)) continue;
-        changed[attr] = val;
+        changed[attr] = val || '';
```
Comment (jridgewell) [inline backbone.js]: Why are we forcing this into a string?

Comment (zgl2007dj) [inline backbone.js]: if changed[a] = undefined
and i post changed to server,  will not work
how to deal with it ? 
` changed[attr] = val || null;`
better?

Comment (jridgewell) [inline backbone.js]: You should normalize in your own code, not in the library.

Comment (akre54) [inline backbone.js]: Agreed.

Comment (zgl2007dj) [inline backbone.js]: @jridgewell 
yeah, I fixed it . 
@akre54 
I had squashed down into a new commit

**Comment by akre54** [comment]
This is still showing up as 7 commits. Try running `git rebase -i HEAD~7`, squash each commit into its previous so you only have one remaining, and then force-push up to github.

**Comment by rafde** [comment]
Doesn't merge also have the option to squash merge now?

**Comment by akre54** [comment]
Yup! They just added it. Gonna test it out now I guess :)

**Pull request merged**